fuse_examples/classification/multimodality/model_tabular_imaging.py

```diff
@@ -0,0 +1,295 @@
+#         self.imaging_inputs = imaging_inputs
+#         self.backbone_categorical_tabular = backbone_categorical_tabular
+#         self.backbone_continuous_tabular = backbone_continuous_tabular
+#         self.backbone_imaging = backbone_imaging
+#         self.projection_imaging = projection_imaging
+#
+#     def forward(self, batch_dict: Dict) -> Dict:
+#
+#         #tabular encoder
+#         categorical_input = FuseUtilsHierarchicalDict.get(batch_dict, self.categorical_tabular_input[0][0])
+#         categorical_embeddings = self.backbone_categorical_tabular(categorical_input)
+#         continuous_input = FuseUtilsHierarchicalDict.get(batch_dict, self.continuous_tabular_input[0][0])
+#         input_cat = torch.cat((categorical_embeddings, continuous_input), 1)
+#         tabular_representations = self.backbone_continuous_tabular(input_cat) #dim 256
+#         FuseUtilsHierarchicalDict.set(batch_dict, 'model.tabular_representations', tabular_representations)
+#
+#         #imaging encoder
+#         imaging_input = FuseUtilsHierarchicalDict.get(batch_dict, self.imaging_inputs[0][0])
+#         backbone_imaging_features = self.backbone_imaging.forward(imaging_input)
+#         res = F.max_pool2d(backbone_imaging_features, kernel_size=backbone_imaging_features.shape[2:])
+#         imaging_representations = self.projection_imaging.forward(res)
+#         imaging_representations = torch.squeeze(imaging_representations)
+#         FuseUtilsHierarchicalDict.set(batch_dict, 'model.imaging_representations', imaging_representations)
+#
+#         return batch_dict['model']
+
+
+
+# concat model
+class TabularImagingConcat(nn.Module):
+    def __init__(self, pooling='max',projection_imaging: nn.Conv2d = nn.Conv2d(384, 256, kernel_size=1, stride=1)):
+        super().__init__()
+        assert pooling in ('max', 'avg')
+        self.pooling = pooling
+        self.projection_imaging = projection_imaging
+
+    def fix_imaging(self,imaging_features):
+        if self.pooling == 'max':
+            imaging_features = F.max_pool2d(imaging_features, kernel_size=imaging_features.shape[2:])
+
+        elif self.pooling == 'avg':
+            imaging_features = F.avg_pool2d(imaging_features, kernel_size=imaging_features.shape[2:])
+
+        imaging_features = self.projection_imaging.forward(imaging_features)
+        imaging_features = torch.squeeze(torch.squeeze(imaging_features,dim=3),dim=2)
+        return imaging_features
+
+
+    def forward(self, batch_dict):
+
+        imaging_features = FuseUtilsHierarchicalDict.get(batch_dict, 'model.imaging_features')
+        tabular_features = FuseUtilsHierarchicalDict.get(batch_dict, 'model.tabular_features')
+        imaging_features = self.fix_imaging(imaging_features)
+        res = torch.cat([tabular_features, imaging_features], dim=1)
+        return res
+
+
+#Tabular model
+class FuseModelTabularContinuousCategorical(torch.nn.Module):
```

Comment:
FYI
In fuse2, I removed the "Fuse" prefix

Reply:
currently supports "current fuse" not fuse 2. will be changed